core/src/main/scala/kafka/log/Log.scala

```diff
@@ -717,16 +748,20 @@ class Log(@volatile var dir: File,
     deleteOldSegments(shouldDelete)
   }
 
+  /*
+   * The log's next offset may be smaller than log start offset after the records in the log are deleted upon DeleteRecordsRequest
```

Comment:
You are right. It is wrong after we add that check. I have removed this comment.